### PR TITLE
UCR pillow refactor in preparation for cross domain UCRs

### PR DESCRIPTION
**Pull request by snopoke**
## Summary
This is a pure refactor. The main goal is to make it easier to customise the pillow behaviour for a new type of UCR data source configuration.

Can be reviewed by commit.

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage
Covered by existing tests

### Safety story
Pure refactor with good test coverage.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

**Comment by snopoke** [comment]
@mkangia @sravfeyn bump

**Review by sravfeyn — APPROVED** [review]

File: corehq/apps/userreports/pillow.py [hunk 1]
```diff
@@ -472,12 +460,6 @@ def __init__(self, processor, pillow_name, topics, num_processes, process_num, r
         # distinguish between doc save errors and data source config errors
         self.retry_errors = retry_errors
 
-    def bootstrap(self, configs=None):
-        self._processor.table_manager.bootstrap(configs)
-
-    def rebuild_table(self, sql_adapter):
-        rebuild_table(sql_adapter)
-
```
Comment (mkangia) [inline corehq/apps/userreports/pillow.py:548]: was this intended in this commit?

File: corehq/apps/userreports/pillow.py [hunk 2]
```diff
@@ -188,6 +188,16 @@ def bootstrap(self, configs=None):
         self.bootstrapped = True
         self.last_bootstrapped = datetime.utcnow()
 
+    @property
+    def relevant_domains(self):
```
Comment (mkangia) [inline corehq/apps/userreports/pillow.py:192]: nit: should this be named `adapters_by_domain`?
`self.table_manager.relevant_domains` makes it look like its going to return domains.

File: corehq/apps/userreports/pillow.py [hunk 3]
```diff
@@ -250,7 +260,7 @@ def _save_doc_to_table(self, domain, table, doc, eval_context):
             table.best_effort_save(doc, eval_context)
         except UserReportsWarning:
             # remove it until the next bootstrap call
-            self.table_manager.table_adapters_by_domain[domain].remove(table)
+            self.table_manager.remove_adapter(domain, table)
```
Comment (mkangia) [inline corehq/apps/userreports/pillow.py:263]: is this removing an adapter or a table? that is a bit confusing here.

Comment (snopoke) [reply corehq/apps/userreports/pillow.py:263]: an adapter

File: corehq/apps/userreports/pillow.py (hunk 1 above)
Comment (snopoke) [reply corehq/apps/userreports/pillow.py:548]: yes, the commit is called "remove unused methods"

File: corehq/apps/userreports/pillow.py (hunk 2 above)
Comment (snopoke) [reply corehq/apps/userreports/pillow.py:192]: it returns a set of domain names

Comment (mkangia) [reply corehq/apps/userreports/pillow.py:192]: Okay, the method is returning `table_adapters_by_domain`, which is possible a dict and you just intend to get the domain which are keys. That makes it clear but is a bit confusing.

File: corehq/apps/userreports/pillow.py (hunk 1 above)
Comment (mkangia) [reply corehq/apps/userreports/pillow.py:548]: oh thanks, i think i read "remove unused imports", my bad.

**Pull request merged**